app/src/main/java/org/dharmaseed/android/TalkRepository.java

```diff
@@ -195,77 +176,37 @@ public Cursor getTalksByCenter(
             long venueId,
             boolean isStarred,
             boolean isDownloaded
-    )
-    {
-        String query = "SELECT talks._id, talks.title, talks.teacher_id, centers.name, talks.rec_date FROM talks";
-
-        if (isStarred)
-        {
-            query += joinStarredTalks();
-        }
-
-        if (isDownloaded)
-        {
-            query += joinDownloadedTalks();
-        }
-
-        query += joinTalkIdVenueId();
-
-        String where = " WHERE ";
-
-        if (searchTerms != null && !searchTerms.isEmpty())
-        {
-            String[] selectionColumns = new String[]
-            {
-                DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.TITLE,
-                DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.DESCRIPTION,
-            };
-
-            where += getSearchStatement(searchTerms, selectionColumns) + " AND ";
-        }
-
-        where += String.format(
-                " %s.%s = %s ",
-                DBManager.C.Talk.TABLE_NAME,
-                DBManager.C.Talk.VENUE_ID,
-                venueId
-        );
-
-        query += where + " ORDER BY talks.rec_date DESC ";
-
-        return queryIfNotNull(query, null);
+    ) {
+        return getTalks(talkAdapterColumns, searchTerms, isStarred, isDownloaded,
+                DBManager.C.Center.TABLE_NAME + "." + DBManager.C.Center.ID + "=" + venueId);
     }
 
 
-    private String joinStarredTalks()
-    {
+    private String joinStarredTalks() {
         return innerJoin(
                 DBManager.C.TalkStars.TABLE_NAME,
                 DBManager.C.TalkStars.TABLE_NAME + "." + DBManager.C.TalkStars.ID,
                 DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.ID
         );
     }
 
-    private String joinDownloadedTalks()
-    {
+    private String joinDownloadedTalks() {
         return innerJoin(
                 DBManager.C.DownloadedTalks.TABLE_NAME,
                 DBManager.C.DownloadedTalks.TABLE_NAME + "." + DBManager.C.DownloadedTalks.ID,
                 DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.ID
         );
     }
 
-    private String joinTalkIdTeacherId()
-    {
+    private String joinTalkIdTeacherId() {
         return innerJoin(
                 DBManager.C.Teacher.TABLE_NAME,
                 DBManager.C.Teacher.TABLE_NAME + "." + DBManager.C.Teacher.ID,
                 DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.TEACHER_ID
         );
     }
 
-    private String joinTalkIdVenueId()
-    {
+    private String joinTalkIdVenueId() {
```

Comment:
Hmm, sorry for all the whitespace noise.  I must've accidentally run android studio's formatter at some point (ctrl+alt+l).  Maybe at some point we can set up a git commit hook to run the formatter to prevent this kind of thing in the future (#66).

Reply:
no worries